### DISCUSSION TITLE
Code generator in v3

### DISCUSSION THREAD

**Discussion by michabbb** [CodeDredd/laravel-soap#223] [Q&A]
hi @CodeDredd,

first, thanks for this amazing package you have created 👏
I have noticed that the part "code generating" is not yet part of version 3. because I am going to start a new soap project, where I usually use https://github.com/WsdlToPhp/PackageGenerator
I wanted to ask what's your advise here to generate all the classes. because https://github.com/wsdl2phpgenerator/wsdl2phpgenerator looks like it's very old and I don't know if it's still maintained.

so what's your advise here please, if someone wants to use your package with full (pre created) support of classes?

thanks a lot!

**Comment by CodeDredd** [comment]
@michabbb 

hi & thanks 😃 
well thats a good question. if i have the time i would rewrite my current code generator to be working without wsdltophp.
The reason you mentioned was also mine why i had to drop code generation in v3 because php 8.1 was no longer supported.

well....my quetion would be....do you really need from start the pregenerated classes? ....or maybe you find the time fixing the code generation in v3 for me ^^....
i am not sure but i think you can also use the pregenerated classes from phpro package. But thats more of a mess up because all the classes are not the "laravel" way.

So from my point of view....i would start without and maybe find time fixing the problem in my package....i can help you there and others maybe love to have that feature back too.

i also wasn't sure how often it was used.

**Reply by michabbb** [reply]
hi,

to be honest, because of the missing time, i wasn´t able to figure out what exactly the benefits are - using your package, instead of using https://github.com/phpro/soap-client directly. so I made my first steps with phpro. my focus is not that much on testing anything here, it´s more about flexibility and I didn´t understand everything in your package, sorry ;)
in my project, I was forced to create my own client with a custom "engine" and I had no idea how all that custom stuff fits into your way of doing things. creating the classes with phpro works totally fine, I personally think, there is no need to copy all that stuff into your own package, which means more maintenance. with all that, i´m sorry, I cannot help you here with your project ;(

but again, I appreciate the work from everybody who supports open-source! THX!

**Reply by CodeDredd** [reply]
Hey,

no need being sorry using https://github.com/phpro/soap-client. I am in contact with the author and if i am not busy and find time i chat with him from time to time.
He is really active and probably if i mention him he will also agree thats its not about what package to choose. Toon does awesome work.
i still need more refactoring inclduing the packages from Toon @veewee.

i am not copying the stuff from toon into my package. I am using his modules and wrapping it around the way laravel works. My client works almost the same way as HTTP which is also only a wrapper around Guzzle.
Like HTTP my SOAP is just a way to make thinks simpler and better for the framework. Using laravel events, etc.
So if you use HTTP from laravel isntead of directly using Guzzle then using your argument it would mean that laravel is producing with HTTP more maintance 😄

**Reply by michabbb** [reply]
thanks for your feedback. as I said, my decision is based on missing knowledge and time working myself into your package. using a wrapper is not a bad thing and I didn't mean it that way. I only wasn't able to see/understand how to bring all my special needs into your wrapper. so I checked the original package and I did understand everything very quickly so that's the reason, there is no "this is bad in general" in the room :)
maybe it's not a bad idea to explain a bit more or show some examples how things are working with your wrapper in the Readme or at some wiki pages, it's only a suggestion :) of course, there will always be someone like me, who has trouble to understand things, so this is just a subjective point of view.

**Reply by CodeDredd** [reply]
you are absoluty right with the documentation part and explaining. For example that i included laravel ray, laraven events, and so on...
i already started making a new one for v3 because the old one is more for v2 and v1.
But as always it is finding time ^^.....but i am on it.

Toon @veewee has too much time. He is like working 12h/day 7 days 🙄

**Reply by veewee** [reply]
Haha, you wish! 🤣
